realsense2_camera/src/named_filter.cpp

```diff
@@ -167,14 +167,16 @@ void PointcloudFilter::Publish(rs2::points pc, const rclcpp::Time& t, const rs2:
 
     rs2_intrinsics depth_intrin = pc.get_profile().as<rs2::video_stream_profile>().get_intrinsics();
 
-    sensor_msgs::PointCloud2Modifier modifier(_msg_pointcloud);
+    sensor_msgs::msg::PointCloud2::UniquePtr _msg_pointcloud = std::make_unique<sensor_msgs::msg::PointCloud2>();
```

Comment:
Do you think that maybe in the initial implemantation reusing the member was more efficiant than creating a new one every frame?
(I am not sure..)

Reply:
BTW, on our naming convention a `_` prefix for a variable indicates a class member.

Reply:
> BTW, on our naming convention a `_` prefix for a variable indicates a class member.

That makes sense. I kept it to keep the variable name but now I removed the prefix.

> Do you think that maybe in the initial implemantation reusing the member was more efficiant than creating a new one every frame?

You may save some memory allocation with this, but the majority of the memory is taken by the dynamically sized `data`. I think it would reallocate memory anyway in some situation, e.g. if the point cloud becomes much larger.

The unique pointer has to be created for every point cloud since it is "moved" and becomes `nullptr` afterwards.